lib/src/state_machine.dart

```diff
@@ -392,7 +389,7 @@ class StateMachine extends Disposable {
 ///     plugIn();
 ///     turnOn(); // "Light is on!"
 ///     unplug(); // "Light is off :("
-class StateTransition extends Disposable implements Function {
```

Comment:
`implements Function`  is a no-op and can be removed.
from https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md#dartcore-1
> The Function type can no longer be implemented, extended or mixed in. Since Dart 2.0 writing implements Function has been allowed for backwards compatibility, but it has not had any effect. In Dart 3.0, the Function type is final and cannot be subtyped, preventing code from mistakenly assuming it works.